### PR TITLE
Add missing actions in Multiselection BottomSheet

### PR DESCRIPTION
**Pull request by LunarX**

### CODE REVIEW

**Review by JorisBodin — APPROVED** [review]

File: app/src/main/res/drawable/ic_non_spam.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (KevinBoulongne) [inline app/src/main/res/drawable/ic_non_spam.xml:1]: Add missing file header.

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/MultiSelectBottomSheetDialog.kt [hunk 2]
```diff
@@ -59,16 +68,64 @@ class MultiSelectBottomSheetDialog : ActionsBottomSheetDialog() {
...
+        }
+    }
+
+    private fun setStateDependentUi(shouldRead: Boolean, shouldFavorite: Boolean) {
+        val readIcon = if (shouldRead) R.drawable.ic_envelope_open else R.drawable.ic_envelope
+        val readText = if (shouldRead) R.string.actionShortMarkAsRead else R.string.actionShortMarkAsUnread
+        binding.mainActions.setAction(R.id.actionReadUnread, readIcon, readText)
+
+        // val isFromArchive = mainViewModel.currentFolder.value?.role == FolderRole.ARCHIVE
+        // TODO : When decided by UI/UX change how the icon is displayed when trying to archive from inside the archive folder
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/MultiSelectBottomSheetDialog.kt:114]:
```suggestion
        // TODO : When decided by UI/UX, change how the icon is displayed (when trying to archive from inside the Archive folder).
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListMultiSelection.kt [hunk 3]
```diff
@@ -227,24 +227,24 @@ class ThreadListMultiSelection {
...
+        private const val ARCHIVE_INDEX = 1
+        private const val FAVORITE_INDEX = 2
 
-            if (!shouldUnRead && !shouldUnFavorite) break
-        }
-        return !shouldUnRead to !shouldUnFavorite
-    }
+        fun computeReadFavoriteStatus(selectedThreads: Set<SelectedThread>): Pair<Boolean, Boolean> {
+            var shouldUnRead = true
+            var shouldUnFavorite = selectedThreads.isNotEmpty()
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListMultiSelection.kt:239]:
```suggestion
            var shouldUnread = true
            var shouldUnfavorite = selectedThreads.isNotEmpty()
```

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/MainActionsView.kt [hunk 4]
```diff
@@ -97,4 +93,26 @@ class MainActionsView @JvmOverloads constructor(
...
+        val drawable = AppCompatResources.getDrawable(context, iconRes)
+        val index = getIndexOfMenuItem(menuId)
+        if (index >= 0) setAction(index, drawable, title)
+    }
+
+    private fun getIndexOfMenuItem(menuId: Int): Int {
+        menu.forEachIndexed { index, item ->
+            if (item.itemId == menuId) return index
+        }
+        return -1
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/MainActionsView.kt:108]:
```suggestion
        }

        return -1
```

**Review by FabianDevel — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/MultiSelectBottomSheetDialog.kt [hunk 5]
```diff
@@ -59,16 +68,64 @@ class MultiSelectBottomSheetDialog : ActionsBottomSheetDialog() {
...
-                    toggleThreadsSpamStatus(selectedThreadsUids)
+                R.id.actionReadUnread -> {
+                    trackMultiSelectActionEvent(ACTION_MARK_AS_SEEN_NAME, selectedThreadsCount, isFromBottomSheet = true)
+                    toggleThreadsSeenStatus(selectedThreadsUids, shouldRead)
+                    isMultiSelectOn = false
+                }
+                R.id.actionArchive -> {
+                    trackMultiSelectActionEvent(ACTION_ARCHIVE_NAME, selectedThreadsCount, isFromBottomSheet = true)
+                    archiveThreads(selectedThreadsUids)
+                    isMultiSelectOn = false
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/MultiSelectBottomSheetDialog.kt:79]: Already done below

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/MultiSelectBottomSheetDialog.kt [hunk 6]
```diff
@@ -59,16 +68,64 @@ class MultiSelectBottomSheetDialog : ActionsBottomSheetDialog() {
                         currentClassName = currentClassName,
                     )
                 }
-                R.id.actionSpam -> {
-                    trackMultiSelectActionEvent(ACTION_SPAM_NAME, selectedThreadsCount, isFromBottomSheet = true)
-                    toggleThreadsSpamStatus(selectedThreadsUids)
+                R.id.actionReadUnread -> {
+                    trackMultiSelectActionEvent(ACTION_MARK_AS_SEEN_NAME, selectedThreadsCount, isFromBottomSheet = true)
+                    toggleThreadsSeenStatus(selectedThreadsUids, shouldRead)
+                    isMultiSelectOn = false
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/MultiSelectBottomSheetDialog.kt:74]: useless

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/MultiSelectBottomSheetDialog.kt [hunk 7]
```diff
@@ -59,16 +68,64 @@ class MultiSelectBottomSheetDialog : ActionsBottomSheetDialog() {
...
+                }
+                R.id.actionArchive -> {
+                    trackMultiSelectActionEvent(ACTION_ARCHIVE_NAME, selectedThreadsCount, isFromBottomSheet = true)
+                    archiveThreads(selectedThreadsUids)
+                    isMultiSelectOn = false
+                }
+                R.id.actionDelete -> {
+                    trackMultiSelectActionEvent(ACTION_DELETE_NAME, selectedThreadsCount, isFromBottomSheet = true)
+                    mainViewModel.deleteThreads(selectedThreadsUids)
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/MultiSelectBottomSheetDialog.kt:83]:
```suggestion
                   deleteThreads(selectedThreadsUids)
```

**Review by sirambd — CHANGES_REQUESTED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Review by JorisBodin — APPROVED** [review]

**Pull request merged**